File: frontend/src/api/workerCare.ts
import http from './http'
import type { ApiEnvelope } from '@/types/api'

export interface WorkerCitation {
  source: string
  article?: string
  title?: string
}

export interface WorkerMessage {
  id: string
  question: string
  answer: string
  answer_source: string
  is_simulated: boolean
  citations?: WorkerCitation[]
  created_at: string
}

export interface TranscribeResult {
  available: boolean
  text: string
  reason: string | null
  provider: string
}

export const workerCareApi = {
  async chat(projectId: string, question: string): Promise<WorkerMessage> {
    const response = await http.post<ApiEnvelope<WorkerMessage>>('/worker-care/chat', { project_id: projectId, question })
    return response.data.data
  },
  async transcribe(projectId: string, audio: Blob): Promise<TranscribeResult> {
    const form = new FormData()
    form.append('project_id', projectId)
    form.append('audio', audio, 'voice.webm')
    const response = await http.post<ApiEnvelope<TranscribeResult>>('/worker-care/transcribe', form)
    return response.data.data
  },
}

File: backend/app/api/v1/endpoints/worker_care.py
from __future__ import annotations

from pydantic import BaseModel, Field
from fastapi import APIRouter, Depends, File, Form, Request, UploadFile
from sqlalchemy.orm import Session

from app.api.dependencies import get_current_user, require_roles
from app.api.response import ok
from app.db.session import get_db
from app.models import User
from app.services.worker_care_service import WorkerCareService

_WORKER_CARE_ROLES = ("admin", "project_manager", "safety_officer", "worker")


class ChatRequest(BaseModel):
    project_id: str
    question: str = Field(min_length=1, max_length=500)


router = APIRouter(prefix="/worker-care", tags=["工友助手"])


@router.post("/chat")
def chat(request: ChatRequest, http_request: Request, user: User = Depends(require_roles(*_WORKER_CARE_ROLES)), db: Session = Depends(get_db)):
    data = WorkerCareService(db).chat(request.project_id, request.question, user)
    return ok(data, http_request, "已生成模板回答")


@router.post("/transcribe")
def transcribe(
    http_request: Request,
    project_id: str = Form(...),
    audio: UploadFile = File(...),
    user: User = Depends(require_roles(*_WORKER_CARE_ROLES)),
    db: Session = Depends(get_db),
):
    data = WorkerCareService(db).transcribe(project_id, audio.file.read(), audio.content_type or "audio/webm", user)
    return ok(data, http_request, "语音转写完成" if data["available"] else "语音转写未配置")
